gobcore/model/sa/gob.py

```diff
@@ -76,29 +98,100 @@ def columns_to_model(table_name, columns):
             # }
 
             # the GOB model for the specified entity
-            fields = {col: desc for col, desc in collection['fields'].items()}
+            entity = _fields_for_collection(collection)
+
+            table_name = model.get_table_name(catalog_name, collection_name)
+            models[table_name] = columns_to_model(table_name, entity)
+
+    return models
+
+
+def _remove_leading_underscore(s: str):
+    return re.sub(r'^_', '', s)
+
 
-            state_fields = STATE_FIELDS if collection.get('has_states') else {}
+def _tablename_from_ref(ref: str):
+    catalog, collection = ref.split(':')
+    return model.get_table_name(catalog, collection)
```

Comment:
Can't we add this method to GOBModel?
The code to split a ref is now duplicated at many places. By placing this code in GOBModel we can prevent further multiplication

Reply:
Done :)